**1. Даны две  независимые выборки. Не соблюдается условие нормальности.**

**x1  380,420, 290**

**y1 140,360,200,900**

**Сделайте вывод по результатам, полученным с помощью функции**

In [1]:
#критерий Манна-Уитни

import numpy as np
import scipy.stats as stats
x1 = np.array([380,420,290])
y1 = np.array([140,360,200,900])
stats.mannwhitneyu(x1,y1)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

Pvalue > alpha, принимаем нулевую гипотезу. В выборках нет статистически значимых различий

**2. Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть ли статистически значимые различия?**

**1е измерение до приема препарата: 150, 160, 165, 145, 155**

**2е измерение через 10 минут: 140, 155, 150,  130, 135**

**3е измерение через 30 минут: 130, 130, 120, 130, 125**

In [2]:
#критерий Фридмана

import numpy as np
import scipy.stats as stats
x1 = np.array([150, 160, 165, 145, 155])
x2 = np.array([140, 155, 150, 130, 135])
x3 = np.array([130, 130, 120, 130, 125])
stats.friedmanchisquare(x1,x2,x3)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

С вероятностью 0.99 есть статистически значимые различия.

**3. Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.**

In [3]:
#критерий Уилкоксона


import numpy as np
import scipy.stats as stats
x1 = np.array([150, 160, 165, 145, 155])
x2 = np.array([140, 155, 150, 130, 135])
stats.wilcoxon(x1,x2)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

При α = 0.0625 и выше принимается альтернативная гипотеза: статистически значимые различия есть, при α < 0.0625 принимается нулевая гипотеза: статистически значимых различий нет.

**4. Даны 3 группы  учеников плавания. В 1 группе время на дистанцию 50 м составляют:**

**56, 60, 62, 55, 71, 67, 59, 58, 64, 67**

**Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53**

**Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54**

**Проверить, есть ли различия между группами**

In [4]:
#критерий Крускала - Уоллиса


import numpy as np
import scipy.stats as stats
gr_1 = np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
gr_2 = np.array([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
gr_3 = np.array([57, 67, 49, 48, 47, 55, 66, 51, 54])
stats.kruskal(gr_1,gr_2,gr_3)

KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)

При α > 0.065 принимается альтернативная гипотеза: статистически значимые различия есть, при α < 0.065 принимается нулевая гипотеза: статистически значимых различий а в результатах разных групп нет.

**5. Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения.**

**бъем выборки 10, уровень статистической значимости 5%**

**2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34**

In [5]:
n = 10
x = [2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34]
m = 2.5
average = sum(x) / len(x)
sigma_n = 0
for i in x:
  sigma_n += (i - average) * (i - average)
sigma_n = pow(sigma_n / (n-1), 0.5)
t_n = (average - m) / (sigma_n / pow(n, 0.5))

print(t_n)

0.5630613661802959


t табличное для 9 степеней свободы и вероятности 0,975 = 2,262. Так как t наблюдаемое больше t табличного, с вероятностью 0,975 верна альтернативная гипотеза. Среднее арифметическое не равно 2,5 см.